stl/inc/memory

```diff
@@ -308,6 +308,20 @@ _NoThrowFwdIt uninitialized_fill_n(_NoThrowFwdIt _First, const _Diff _Count_raw,
 }
 #endif // _HAS_IF_CONSTEXPR
 
+#if _HAS_CXX20
+template <class _Ty, class... _Args_pack>
+/* constexpr */ _Ty* construct_at(_Ty* const _Location, _Args_pack&&... _Args) {

```

Comment:
I believe that you can already "implement" a lot of it similar to `_CONSTEXPR_ICE` with a macro that we can switch once compiler support is ready.

Reply:
This PR is really only about `construct_at` (because I really like this function). I don't want to play with constexpr new yet, since I don't know when msvc will support it and whether it would make merging this PR more difficult. If adding `_CONSTEXPR_NEW` wouldn't make this PR more complicated I can obviously implement it.

Reply:
I believe you can simply write something like this:

```cpp
#ifdef __cpp_lib_constexpr_dynamic_alloc 
#define _CONSTEXPR_NEW constexpr
#else // ^^^__cpp_lib_constexpr_dynamic_alloc / !__cpp_lib_constexpr_dynamic_alloc  vvv
#define _CONSTEXPR_NEW 
#endif // !__cpp_lib_constexpr_dynamic_alloc 
```

Then you could apply that to everything from #37 that depends on constexpr new and be done with it until the compiler feature is there.

Reply:
Implementing it would be easy but it might require that maintainers make some test coverage for this which would be impossible without compiler supporting this feature. I would prefer this PR to be merged quickly if possible and not be blocked on compiler update.

Reply:
I mean the whole reason is that there is no compiler implementing it so that we can be sure that `_CONSTEXPR_NEW` evaluates to nothing. On the other hand things like `/* constexpr */ ` require extensive maintainance afterwards. 

If you want to be safe can also simply go with the following

```cpp
#if 0 // TRANSITION: #ifdef __cpp_lib_constexpr_dynamic_alloc 
#define _CONSTEXPR_NEW constexpr
#else // ^^^__cpp_lib_constexpr_dynamic_alloc / !__cpp_lib_constexpr_dynamic_alloc  vvv
#define _CONSTEXPR_NEW 
#endif // !__cpp_lib_constexpr_dynamic_alloc
```

Taht way you only need to change a single line rather than searching through all those comments.

Reply:
Ok I will go with `if 0` and if maintainers are ok with `ifdef __cpp_constexpr_dynamic_alloc` I will change it.